src/Uno.UI.RemoteControl/HotReload/ClientHotReloadProcessor.MetadataUpdate.cs

```diff
@@ -170,56 +171,108 @@ static int[] ReadIntArray(BinaryReader binaryReader)
 			return values;
 		}
 #endif
-		private static void ReloadWithUpdatedTypes(Type[] updatedTypes)
+
+		private static int _isReloading;
+
+		private static async void ReloadWithUpdatedTypes(Type[] updatedTypes)
 		{
-			if (updatedTypes.Length == 0)
+			if (_log.IsEnabled(LogLevel.Trace))
+			{
+				_log.Trace($"ReloadWithUpdatedTypes - start");
+			}
+
+			if (Interlocked.CompareExchange(ref _isReloading, 1, 0) == 1)
 			{
 				return;
 			}
```

Comment:
This feels like it should be in the `TypeMappingHelper`, and make the `WaitToHotReload` return async `true` only for one of the waiters of the method (so only one is actually allowed to perform the reload).

Reply:
I've adjusted `TypeMappingHandler` (now just `TypeMappings`) so that the methods are specifically about the collection of type mappings. This includes renaming `WaitToHotReload` to `WaitForMappingsToResume`. I don't think it makes sense to move this logic inside the TypeMappingHandler as it's specifcally about ensuring there's only one (potentially blocked) UI update. I'll adjust the logic here to clean it up a bit